prog/postgres/postgres_resource_nexus.rb

```diff
@@ -25,7 +25,14 @@ def self.assemble(project_id:, location:, server_name:, target_vm_size:, target_
       superuser_password, timeline_id, timeline_access = if parent_id.nil?
         [SecureRandom.urlsafe_base64(15), Prog::Postgres::PostgresTimelineNexus.assemble.id, "push"]
       else
-        parent = PostgresResource[parent_id]
+        unless (parent = PostgresResource[parent_id])
+          fail "No existing parent"
+        end
+        restore_target = Validation.validate_date(restore_target, "restore_target")
+        unless parent.timeline.earliest_restore_time && parent.timeline.earliest_restore_time <= restore_target &&
```

Comment:
It is low probability, but we should refresh the earliest_restore_time before using it in a comparison, in case the earliest backup is deleted before user submits the form.

Reply:
I added `refresh` parameter to `earliest_restore_time`, and call `refresh: true` here

```ruby
def earliest_restore_time(refresh: false)
  if (earliest_backup = ((!refresh && earliest_backup_completed_at) || refresh_earliest_backup_completion_time))
    earliest_backup + 5 * 60
  end
end
```

Reply:
I'd just call refresh_earliest_backup_completion_time before accessing earliest_restore_time because that would result in slightly more readable earliest_restore_time. Also boolean arguments sometimes seen as code smell [[1](https://martinfowler.com/bliki/FlagArgument.html)], but in this case it seems OK.

Both options are fine IMO.